rasa/shared/utils/validation.py

```diff
@@ -242,6 +242,7 @@ def validate_training_data_format_version(
         return True
 
     try:
+        version_value = version_value.strip("\"'")
```

Comment:
Why is this required? Also causes a unit test to fail 🤔

Reply:
because the version tags in the configs will look like `version: '3.0'` and the yaml read function will convert that to `"'3.0'"` (if we just write `3.0`, then we get an int and a type check will fail instead)